### PR TITLE
Check graveyard on dangling index import

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today if you concurrently import a dangling index twice and delete it
then you can end up with a cluster state that holds the index as well as
its tombstone, which is invalid. This commit addresses that by ensuring
that the index is definitely not in the graveyard before importing it.

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java [hunk 1]
```diff
@@ -271,6 +276,70 @@ public void testDeleteDanglingIndicesRequiresDataLossFlagToBeTrue() throws Excep
...
+
+        final String stoppedNodeName = createDanglingIndices(INDEX_NAME, OTHER_INDEX_NAME);
+        final String danglingIndexUUID = findDanglingIndexForNode(stoppedNodeName, INDEX_NAME);
+
+        final AtomicBoolean isImporting = new AtomicBoolean();
+        final Thread[] importThreads = new Thread[2];
+        for (int i = 0; i < importThreads.length; i++) {
+            importThreads[i] = new Thread(() -> {
+                //noinspection StatementWithEmptyBody
+                while (isImporting.get() == false) {
```
Comment (henningandersen) [inline server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java:295]: nit: I prefer a latch just to not burn cycles on CI unnecessarily. Also, this avoids the very unlikely edge case of the main thread passing through 10 seconds of trying to delete the index either before the thread gets here or in one of the loop iterations.

**Review by henningandersen — APPROVED** [review]
LGTM, nice catch.

**Review by ywelsch — APPROVED** [review]
LGTM

File: server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java]: :+1: see https://github.com/elastic/elasticsearch/pull/67687/commits/22859605916e82a54148a23acde11f60b5dffc54

**Pull request merged**